pytorch_lightning/callbacks/batch_size_finder.py

```diff
@@ -0,0 +1,423 @@
+
+        self._early_exit = False
+
+        from pytorch_lightning.loggers.base import LightningLoggerBase
+
+        class _BatchSizeFinderDumpedParams(TypedDict):
+            callbacks: List[Callback]
+            logger: Optional[LightningLoggerBase]
+            max_steps: int
+            global_step: Optional[None]
+            limit_val_batches: Union[int, float]
+            limit_eval_batches: Union[int, float]
+
+        self._dumped_params: _BatchSizeFinderDumpedParams = {}
+
+    def scale_batch_size(self, trainer: "pl.Trainer", pl_module: "pl.LightningModule") -> None:
+        if trainer.fast_dev_run:
+            rank_zero_warn("Skipping batch size scaler since `fast_dev_run` is enabled.")
+            return
+
+        if not lightning_hasattr(pl_module, self.batch_arg_name):
+            raise MisconfigurationException(
+                f"Field {self.batch_arg_name} not found in both `model` and `model.hparams`"
+            )
+
+        if (
+            hasattr(pl_module, self.batch_arg_name)
+            and hasattr(pl_module, "hparams")
+            and self.batch_arg_name in pl_module.hparams
+        ):
+            rank_zero_warn(
+                f"Field `model.{self.batch_arg_name}` and `model.hparams.{self.batch_arg_name}` are mutually exclusive!"
+                f" `model.{self.batch_arg_name}` will be used as the initial batch size for scaling."
+                " If this is not the intended behavior, please remove either one."
+            )
+
+        # TODO: check if this can be enabled (#4040)
+        if not trainer._data_connector._train_dataloader_source.is_module():
+            raise MisconfigurationException(
+                "The batch scaling feature cannot be used with dataloaders passed directly to `.fit()`."
+                " Please disable the feature or incorporate the dataloader into the model."
+            )
+
+        # Arguments we adjust during the batch size finder, save for restoring
+        self._dump_params(trainer)
+
+        # Set to values that are required by the algorithm
+        self._reset_params(trainer)
+
+        # Save initial model, that is loaded after batch size is found
+        save_path = os.path.join(trainer.default_root_dir, f".scale_batch_size_temp_model_{uuid.uuid4()}.ckpt")
+        trainer.save_checkpoint(save_path)
+
+        if trainer.progress_bar_callback:
+            trainer.progress_bar_callback.disable()
+
+        new_size, _ = self._adjust_batch_size(trainer, value=self.init_val)
+
+        if self.mode == "power":
+            new_size = self._run_power_scaling(trainer, pl_module, new_size)
+        elif self.mode == "binsearch":
+            new_size = self._run_binary_scaling(trainer, pl_module, new_size)
+
+        garbage_collection_cuda()
+
+        if trainer.is_global_zero:
+            trainer.checkpoint_connector.restore(save_path)
+            fs = get_filesystem(save_path)
+            if fs.exists(save_path):
```

Comment:
Is this required as you have created the file before ?

Reply:
Should this be within the teardown in case the tuner fails in the process and the checkpoint requires to be cleaned out ?

Reply:
if tuner fails in the process, then we handle that case within callback, so this will run eventually.